app/lib/shared/datastore_helper.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:gcloud/db.dart';
+import 'package:retry/retry.dart';
+
+class TransactionWrapper {
+  final Transaction _tx;
+  bool _mutated = false;
+
+  TransactionWrapper._(this._tx);
+
+  Future<List<T>> lookup<T extends Model>(List<Key> keys) =>
+      _tx.lookup<T>(keys);
+
+  Future<T> lookupOrNull<T extends Model>(Key key) =>
+      _tx.lookupValue<T>(key, orElse: () => null);
+
+  Future<T> lookupValue<T extends Model>(Key key, {T orElse()}) =>
+      _tx.lookupValue<T>(key, orElse: orElse);
+
+  Query<T> query<T extends Model>(Key ancestorKey, {Partition partition}) =>
+      _tx.query<T>(ancestorKey, partition: partition);
+
+  void insert(Model entity) => queueMutations(inserts: [entity]);
+
+  void delete(Key key) => queueMutations(deletes: [key]);
+
+  void queueMutations({List<Model> inserts, List<Key> deletes}) {
+    _mutated = true;
+    _tx.queueMutations(inserts: inserts, deletes: deletes);
+  }
+}
+
+/// Call [fn] with a [TransactionWrapper] that is either committed or
+/// rolled back when [fn] returns.
+Future<T> withTransaction<T>(
+  DatastoreDB db,
+  Future<T> Function(TransactionWrapper) fn,
```

Comment:
This could be `Future<T> Function(Transaction) fn`

Reply:
Yes, but `Transaction` exposes `commit()` and `rollback()` which this function calls on your behalf.